micro-rdk/src/common/conn/viam.rs

```diff
@@ -489,14 +489,21 @@ where
         }
 
         // Since provisioning was run and completed, credentials are properly populated
-        // if wifi manager is configured loop forever until wifi is connected
+        // if wifi manager is configured loop forever until wifi is connected via
+        // a the provisioned network or one from previously stored agent config
         if let Some(wifi) = self.wifi_manager.as_ref().as_ref() {
-            while let Err(err) = wifi
-                .set_sta_mode(self.storage.get_wifi_credentials().unwrap())
-                .await
-            {
-                log::error!("couldn't connect to wifi reason {:?}", err);
-                let _ = Timer::after(Duration::from_secs(2)).await;
+            let mut networks = self.storage.get_all_networks().unwrap();
+            networks.sort_by(|a, b| b.priority.cmp(&a.priority));
+            log::debug!("networks to try: {:?}", networks);
+            for network in networks.iter().cycle() {
+                log::info!("attempting to connect to network `{}`", network.ssid);
+                if let Err(err) = wifi.set_sta_mode(network.clone()).await {
+                    log::error!("failed to connect to network: {:?}", err);
+                    let _ = Timer::after(Duration::from_secs(1)).await;
+                } else {
+                    log::info!("successfully connected to network `{}`", network.ssid);
+                    break;
+                }
```

Comment:
If we have a thing called `WifiManager`, I'd expect to see this sort of logic encapsulated there, rather than explicitly written out here.

Reply:
Another reason to like WifiManager for this is that it has (I think) the ability to scan for visible wifi networks. That'd let you filter out any in the list that aren't even visible, making for a faster connection experience overall.